File: typescript/src/messages/verdict_request_for_url.ts
import { JsonProperty, JsonObject } from "typescript-json-serializer";
import { Kind, Message } from "./message";

@JsonObject()
export class VerdictRequestForUrl extends Message {
  public constructor(url: URL, guid: string, session_id: string, use_cache: boolean | undefined, use_shed: boolean | undefined) {
    super(Kind.VerdictRequestForUrl);
    this.url = url.toString();
    this.session_id = session_id;
    this.guid = guid;
    this.use_cache = use_cache;
    this.use_hash_lookup = use_shed;
  }

  @JsonProperty() public url: string;
  @JsonProperty() public guid: string;
  @JsonProperty() public session_id: string;
  @JsonProperty() public use_hash_lookup: boolean | undefined;
  @JsonProperty() public use_cache: boolean | undefined;  
}

File: rust/src/message/verdict_request_for_url.rs
use super::VerdictRequest;
use crate::message::kind::Kind;
use reqwest::Url;
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct VerdictRequestForUrl {
    pub url: String,
    pub kind: Kind,
    pub guid: String,
    pub session_id: String,
    pub use_shed: bool,
    pub use_cache: bool,
}

impl VerdictRequestForUrl {
    pub fn new(url: &Url, session_id: String, use_cache: bool, use_shed: bool) -> Self {
        Self {
            guid: uuid::Uuid::new_v4().to_string(),
            url: url.to_string(),
            kind: Kind::VerdictRequestForUrl,
            session_id,
            use_cache,
            use_shed,
        }
    }
}

impl VerdictRequest for VerdictRequestForUrl {
    fn guid(&self) -> &str {
        &self.guid
    }
}
